jormungandr/src/network/subscription.rs

```diff
@@ -20,136 +19,74 @@ pub fn process_block_announcements<S>(
     node_id: NodeId,
     global_state: GlobalStateR,
     block_box: MessageBox<BlockMsg>,
-    logger: Logger,
+    logger: &Logger,
 ) where
     S: Stream<Item = Header, Error = core_error::Error> + Send + 'static,
 {
-    let state = global_state.clone();
+    let sink = BlockAnnouncementProcessor::new(block_box, node_id, global_state.clone(), logger);
+    let logger = sink.logger.clone();
     let stream_err_logger = logger.clone();
-    let sink_err_logger = logger.clone();
-    let stream = inbound
-        .map_err(move |err| {
-            debug!(
-                stream_err_logger,
-                "block subscription stream failure: {:?}", err
-            );
-        })
-        .map(move |header| {
-            state.peers.refresh_peer_on_block(node_id);
-            BlockMsg::AnnouncedBlock(header, node_id)
-        });
-    global_state.spawn(
-        block_box
-            .sink_map_err(move |_| {
-                error!(
-                    sink_err_logger,
-                    "failed to send block announcement to the block task"
-                );
-            })
-            .send_all(stream)
-            .map(move |_| {
-                debug!(logger, "block subscription ended");
-            }),
-    );
-}
-
-pub fn process_block_announcement(
-    header: Header,
-    node_id: NodeId,
-    global_state: &GlobalState,
-    block_box: MessageBox<BlockMsg>,
-) -> SendingBlockMsg {
-    global_state.peers.refresh_peer_on_block(node_id);
-    let future = block_box.send(BlockMsg::AnnouncedBlock(header, node_id));
-    SendingBlockMsg { inner: future }
-}
-
-#[must_use = "futures do nothing unless polled"]
-pub struct SendingBlockMsg {
-    inner: sink::Send<MessageBox<BlockMsg>>,
-}
-
-impl Future for SendingBlockMsg {
-    type Item = MessageBox<BlockMsg>;
-    type Error = core_error::Error;
-
-    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
-        self.inner.poll().map_err(|_e| {
-            core_error::Error::new(
-                core_error::Code::Aborted,
-                "the node stopped processing blocks",
-            )
-        })
-    }
+    let stream = inbound.map_err(move |e| {
+        debug!(
+            stream_err_logger,
+            "block subscription stream failure";
+            "error" => ?e,
```

Comment:
```suggestion
            "reason" => ?e,
```

just so we remain consistent?

Reply:
That's inconsistent by me :grin: I've been using `"error" => ?e` in debug logs and `"reason" => %e` in info logs which should produce more user-friendly messages.

I think it's time to deliberate and lay down the law on this.